File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min =prices[0];
        int ndiff=0;
        for (int i=1;i<prices.length;i++) {
            if (prices[i]<min) {
                min=prices[i];
            } else {
                int diff=prices[i]-min;
                if (diff>ndiff) 
                {
                    ndiff=diff;
                }
            }
        }
        return ndiff;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min = float('inf')
        max = 0
        diff = 0
        ndiff = 0

        for i in range(len(prices)):
            if prices[i] < min:
                min = prices[i]
            if prices[i] - min > diff:
                diff = prices[i] - min
            if diff > ndiff:
                ndiff = diff
                
        return ndiff
